### ISSUE TITLE
[terra-clinical-header] Accessibility Updates

### ISSUE THREAD

**Issue by saket2403** [cerner/terra-clinical#791]
# Feature Request

## Description
As a consumer of Terra-Clinical-Header, I would like to the user to be able to select a heading level, so that it does not have accessibility assumptions or defaults that create incorrect context for users and my application meets [WCAG Success Criterion 1.3.1 (Info and Relationships)](https://www.w3.org/TR/WCAG21/#info-and-relationships) plus Success Criterion 1.3.2 (Meaningful Sequence).
When headings [reflect page organization](https://www.w3.org/WAI/tutorials/page-structure/headings/#headings-that-reflect-the-page-organization), they should be given ‘id’s and used with ‘aria-labelledby’ to associate headings with the corresponding page region​
example: https://www.w3.org/WAI/tutorials/page-structure/example/

**Comment by saket2403** [comment]
# Tech Design
## Summary:
Add `level`, `text` and `id` props to terra-clinical-header to enhance the accessibility across ultiple browser-screenreader combos.

## Requirements and Accessibility:
As a consumer of Terra-Clinical-Header, I would like to the user to be able to select a heading level, so that it does not have accessibility assumptions or defaults that create incorrect context for users and my application meets [WCAG Success Criterion 1.3.1 (Info and Relationships)](https://www.w3.org/TR/WCAG21/#info-and-relationships) plus Success Criterion 1.3.2 (Meaningful Sequence).
When headings [reflect page organization](https://www.w3.org/WAI/tutorials/page-structure/headings/#headings-that-reflect-the-page-organization), they should be given ‘id’s and used with ‘aria-labelledby’ to associate headings with the corresponding page region​

## React Props:
### Terra-clinical-header
| Prop | Type | isRequired | Default | Description |
|----------|------|--------------|---------|--------------|
| text | string | required | N/A | Text to be displayed as the title in the header bar |
| level | enum:[1, 2, 3, 4, 5, 6] | optional | undefined |  Sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. This helps screen readers to announce appropriate heading levels. Changing 'level' will not visually change the style of the content. |
| id | String | optional | undefined | IDs should be used with ‘aria-labelledby’ to associate headings with the corresponding page region. |

## Usage Example

```javascript
    const header= <Header text="Test-Header" id="Header" level={1} />;
    <ContentContainer aria-labelledby="Header" id="content-container" header={header}>
      <div>content</div>;
    </ContentContainer>
```

**Comment by benbcai** [comment]
Fixed by #792
